modules/@apostrophecms/schema/ui/apos/logic/AposInputArray.js

```diff
@@ -247,14 +371,47 @@ export default {
       const item = this.items.find(item => item._id === id);
       return get(item.schemaInput.data, titleField) || `Item ${index + 1}`;
     },
-    openInlineItem(id) {
-      this.items.forEach(item => {
-        item.open = (item._id === id) || this.alwaysExpand;
-      });
+    toggleOpenInlineItem(event) {
+      if (event) {
+        const el =
+          event.target.getAttribute('data-id')
+            ? event.target
+            : event.target.closest('[data-id]');
+        const elId = el.getAttribute('data-id');
+        if (this.items.find(i => i._id === elId)) {
+          event.preventDefault();
+          const was = this.items.find(i => i._id === elId).open;
+          this.items.find(i => i._id === elId).open = !was;
+        }
+      }
     },
-    closeInlineItem(id) {
-      this.items.forEach(item => {
-        item.open = this.alwaysExpand;
+    moveEngaged(event, id, direction) {
+      const item = this.items.find(i => i._id === id);
+      const index = this.items.indexOf(item);
+
+      if (
+        ((index + direction) === this.items.length) ||
+        ((index + direction < 0))
+      ) {
+        // already first or last, don't move
+        return;
+      }
+
+      if (this.items.find(i => i._id === id).engaged) {
+        event.stopImmediatePropagation();
+        event.preventDefault();
```

Comment:
Can use event modifiers on directives directly:
```html
@keydown.stop.prevent.arrow-up
```
Not sure this is what you need here though.

Reply:
got these properly prevented at the ui level